snapshot-labs/keycard
# feat: New tiers

Related to https://github.com/snapshot-labs/pitches/issues/43

TODO:
- [x] Run this on keycard prod DB
  ```SQL
  ALTER TABLE
    `keys`
  ADD COLUMN
    `tier` TINYINT NOT NULL DEFAULT 0 AFTER name;
  ```

### Summary of changes:
- New level column in keys table
- API will now return a new param `key_counts` which will contain all active keys with values of level and count
- Docker MySQL port change to avoid conflicts with local MySQL (should not break anything)
- Fix tests

## How to test:
- Run `docker compose up --build`
- Make sure API is up and running
- Call `get_keys` method https://github.com/snapshot-labs/keycard#get_keys
- Notice the updated `limits` and `key_counts` in response
